Sources/Testing/Issues/Confirmation.swift

```diff
@@ -190,3 +178,52 @@ public func confirmation<R>(
   }
   return try await body(confirmation)
 }
+
+@_spi(Experimental)
+extension Confirmation {
+  /// A protocol that describes a range expression that can be used with
+  /// ``confirmation(_:expectedCount:sourceLocation:_:)-41gmd``.
+  ///
+  /// This protocol represents any expression that describes a range of
+  /// confirmation counts. For example, the expression `1 ..< 10` automatically
+  /// conforms to it.
+  ///
+  /// You do not generally need to add conformances to this type yourself. It is
+  /// used by the testing library to abstract away the different range types
+  /// provided by the Swift standard library.
+  public protocol ExpectedCount: Sendable, RangeExpression<Int> {}
+}
+
+extension Confirmation.ExpectedCount {
+  /// Get an instance of ``Issue/Kind-swift.enum`` corresponding to this value.
+  ///
+  /// - Parameters:
+  ///   - actualCount: The actual count for the failed confirmation.
+  ///
+  /// - Returns: An instance of ``Issue/Kind-swift.enum`` that describes `self`.
+  fileprivate func issueKind(forActualCount actualCount: Int) -> Issue.Kind {
+    switch self {
+    case let expectedCount as ClosedRange<Int> where expectedCount.lowerBound == expectedCount.upperBound:
+      return .confirmationMiscounted(actual: actualCount, expected: expectedCount.lowerBound)
+    case let expectedCount as Range<Int> where expectedCount.lowerBound == expectedCount.upperBound - 1:
```

Comment:
Nit:

Considering 5..<5 is a valid range expression.

If someone is passing `Int.min ..< Int.min`, this will cause a "Swift runtime failure: arithmetic overflow".

We can just use `&-` here to avoid the corner case crash. (But I guess probably no one would pass `Int.min ..< Int.min` as expectedCount here LOL)

```suggestion
    case let expectedCount as Range<Int> where expectedCount.lowerBound == expectedCount.upperBound &- 1:
```

Reply:
`n ..< n` is not a valid range expression for integer types as the upper bound (effectively `n - 1`) is less than the lower bound.

Reply:
```
Swift/arm64e-apple-macos.swiftinterface:22083: Fatal error: Range requires lowerBound <= upperBound
```

Swift Standard library will accept the form eg. lowerBound ..< upperBound.

Do you mean we have a more strict rule for Int range and the standard library has not enforce such rule?

Reply:
It may _accept_ construction of such an expression, but it will result in an invalid range. As well, a constraint that a condition be met fewer than zero times cannot be satisfied in reality, so passing such a range constitutes a logic error in your test.

Reply:
Yes. I agree such input is an invalid input. And the behavior shall be undefined.

We can use `&-` to make it not crash and we can also choose do nothing and make it crash.

Reply:
What I'm saying is that it is _correct_ that such code crashes, because an upper bound of `.min` is never going to be valid in the first place. 🙂

If we wanted to be pedantic, we could add a protocol requirement to `ExpectedCount` of the form `func validate() throws` that, for a half-open range like `n ..< m`, tests that `n < m && m >= 0`, but that's overdesigning the feature for the sake of a corner case that, again, will never function correctly in the first place.